kolibri/core/content/views.py

```diff
@@ -290,7 +290,11 @@ def get(self, request, zipped_filename, embedded_filepath):
         zipped_path = get_path_or_404(zipped_filename)
 
         # Sometimes due to URL concatenation, we get URLs with double-slashes in them, like //path/to/file.html.
-        # Normalize the path by converting those double-slashes to a single slash.
+        # If it is a leading double slash (i.e. /zipcontent/filename.zip//file.html) then it will get passed
+        # to zipcontent as a single leading slash - detect this and remove if present.
+        if embedded_filepath.startswith("/"):
+            embedded_filepath = embedded_filepath[1:]
```

Comment:
This is confusing. The comment refers to a leading double slash, but

* the example shows a double slash later in the path
* the test looks for a single leading slash

Reply:
Maybe the comment can be improved, but the reason we need to handle this special case is because `embedded_filename` has already had the leading slash removed by Django as part of its built-in URL parsing. (See the `/zipcontent` URLs formatting string for more details) Hence for the URL `/zipcontent/filename.zip//file.html`, when `get` is called, `embedded_filename` will have the value `/file.html`, and thus will not be a relative path, but it also won't be a double-slash so it won't be caught by the original handling.

Reply:
Ah, I see. I didn't look closely enough at it to notice `filename.zip` in the middle

Reply:
I am happy to accept suggestions for how to rephrase the comment :)